include/systems/eigen_system.h

```diff
@@ -178,12 +168,12 @@ class EigenSystem : public System
   /**
    * The system matrix for standard eigenvalue problems.
    */
-  std::unique_ptr<SparseMatrix<Number>> matrix_A;
+  SparseMatrix<Number> * matrix_A;
```

Comment:
I fear this leads to screaming when matrix_A.get() in user code stops working.  I don't see any good way around it, though.

Reply:
In hindsight... it would have been good to have these behind accessors rather than public members... if we're breaking backwards-compatibility anyway, maybe we should add accessors? Not really sure what's best... at least now someone can't "steal" the matrix by doing
```
auto A = eigen_sys.matrix_A;
```
so already an improvement IMO.

Reply:
The catch is that we're only *partly* breaking backwards-compatibility.  Anyone accessing the matrices solely through operator-> (or through operator* and operations on the reference) is still fine; it's just people trying to pass-by-pointer who are in trouble.

I'd love to see accessors for these, even if it's kind of too-little too-late, though.  Then give it 8 or 9 months and we can give the raw pointers private access specifiers whenever ifndef LIBMESH_ENABLE_DEPRECATED.